src/core/a-entity.js

```diff
@@ -704,29 +690,33 @@ var proto = Object.create(ANode.prototype, {
                              HTMLElement.prototype.getAttribute.call(this, attrName));
       }
 
-      // Determine which type of setAttribute to call based on the types of the arguments.
-      if (arg1Type === 'string' && typeof arg2 !== 'undefined') {
-        singlePropertyUpdate(this, attrName, arg1, arg2);
+      // Determine new attributes from the arguments
+      if (
+        typeof arg2 !== 'undefined' &&
+        typeof arg1 === 'string' &&
+        arg1.length > 0 &&
+        (typeof utils.styleParser.parse(arg1)) === 'string'
```

Comment:
I ordered the clauses of the `if` to short circuit as favorably as possible, so that the overhead of the final clause is only encountered in the case where it is unavoidable: you've passed 3 arguments, and the 2nd is a non-empty string.

Because the single property update case has the most restrictions on its arguments, I think it is probably more elegant to test for.

If you *are* passing 3 arguments with the 2nd a string, the overhead must exist unless we move the logic of differentiation down to the component (where the parsed data can be reused) or redesign the API to not be ambiguous for strings (e.g. not accept a property name in the 2nd parameter at all).